src/models/eventmodel.cpp

```diff
@@ -554,11 +605,20 @@ QModelIndex EventModel::parent(const QModelIndex& child) const
         return createIndex(OverViewRow::Process_Row, 0, static_cast<quintptr>(Tag::Overview));
     case Tag::Tracepoints:
         return createIndex(OverViewRow::Tracepoint_Row, 0, static_cast<qintptr>(Tag::Overview));
-    case Tag::Threads: {
+    case Tag::Favorites:
+        return createIndex(OverViewRow::Favorite_Row, 0, static_cast<qintptr>(Tag::Overview));
+    case Tag::Threads:
         const auto parentRow = tagData(child.internalId());
         return createIndex(parentRow, 0, static_cast<quintptr>(Tag::Processes));
     }
-    }
 
     return {};
 }
+
+void EventModel::addToFavourites(const QModelIndex& index)
+{
+    const int position = m_favourites.size();
```

Comment:
assert that the index.model() == this _and_ that `.column() == 0`

return early if the index is in the list of favorites already